Intermittently failing MapIndexLifecycleTest

com.hazelcast.map.MapIndexLifecycleTest.whenIndexConfigured_existsOnAllMembers

**Stacktrace**
```
java.lang.AssertionError: Year index should contain records
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.map.MapIndexLifecycleTest.assertGlobalIndexesAreInitialized(MapIndexLifecycleTest.java:181)
	at com.hazelcast.map.MapIndexLifecycleTest.whenIndexConfigured_existsOnAllMembers(MapIndexLifecycleTest.java:116)
```

https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/14376/testReport/junit/com.hazelcast.map/MapIndexLifecycleTest/whenIndexConfigured_existsOnAllMembers/

**Standard Output** 
[test.log](https://github.com/hazelcast/hazelcast/files/1811535/test.log)

--- a/hazelcast/src/test/java/com/hazelcast/map/MapIndexLifecycleTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/map/MapIndexLifecycleTest.java
@@ -17,8 +17,6 @@
 package com.hazelcast.map;
 
 import com.hazelcast.config.Config;
-import com.hazelcast.config.InMemoryFormat;
-import com.hazelcast.config.MapConfig;
 import com.hazelcast.config.MapIndexConfig;
 import com.hazelcast.config.MapStoreConfig;
 import com.hazelcast.config.ServiceConfig;
@@ -177,9 +175,13 @@ public class MapIndexLifecycleTest extends HazelcastTestSupport {
             }
         });
 
-
-        assertTrue("Year index should contain records",
-                mapContainer.getIndexes().getIndex("year").getRecords(1801).size() > 0);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertTrue("Year index should contain records",
+                        mapContainer.getIndexes().getIndex("year").getRecords(1801).size() > 0);
+            }
+        });
     }
 
     private int numberOfPartitionQueryResults(OperationService operationService, int partitionId,

